src/procedures/__tests__/TransferErc20.ts

```diff
@@ -0,0 +1,203 @@
+      'Erc20TokenBalanceFactory'
+    );
+
+    factoryMockSetup = mockFactories();
+    factoryMockSetup.erc20TokenBalanceFactory = erc20TokenBalanceFactoryMock.getMockInstance();
+
+    erc20TokenBalanceFactoryMock.mock('refresh', Promise.resolve());
+    contextMock.set('factories', factoryMockSetup);
+    contextMock.set('currentWallet', new Wallet({ address: () => Promise.resolve(currentWallet) }));
+
+    polyTokenMock = ImportMock.mockClass(contractWrappersModule, 'PolyToken');
+    polyTokenMock.mock('address', Promise.resolve(polyTokenAddress));
+    wrappersMock.set('polyToken', polyTokenMock.getMockInstance());
+    wrappersMock.mock('isTestnet', Promise.resolve(false));
+
+    // Instantiate TransferErc20
+    target = new TransferErc20(params, contextMock.getMockInstance());
+  });
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have TransferErc20 type', async () => {
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe(ProcedureType.TransferErc20);
+    });
+  });
+
+  describe('TransferErc20', () => {
+    test('should add a transaction to the queue to transfer an erc20 token with specified token address to a specified receiving address', async () => {
+      const addTransactionSpy = spy(target, 'addTransaction');
+      erc20Mock.mock('transfer', Promise.resolve('Transfer'));
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(addTransactionSpy.getCall(0).calledWith(erc20Mock.getMockInstance().transfer)).toEqual(
+        true
+      );
+      expect(addTransactionSpy.getCall(0).lastArg.tag).toEqual(PolyTransactionTag.TransferErc20);
+      expect(addTransactionSpy.callCount).toEqual(1);
+    });
+
+    test('should throw if supplied address does not correspond to a valid erc20 token', async () => {
+      wrappersMock
+        .mock('getERC20TokenWrapper')
+        .withArgs({ address: params.tokenAddress })
+        .throws();
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: 'The supplied address does not correspond to an ERC20 token',
+        })
+      );
+    });
+
+    test('should throw if address belongs to a security token, not an erc20 token', async () => {
+      securityTokenRegistryMock.mock('isSecurityToken', Promise.resolve(true));
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message:
+            "This address belongs to a Security Token. To transfer Security Tokens, use the functions in the Security Token's transfers namespace",
+        })
+      );
+    });
+
+    test('should add a transaction to the queue to transfer an erc20 token getting poly token funds if there are not enough funds, only on testnet', async () => {
```

Comment:
The description is confusing. What is being transferred is actually POLY in this case.

It should say something like 'should add an extra transaction to get POLY from the faucet if the balance is insufficient on testnet'